numpy/core/src/umath/ufunc_type_resolution.c

```diff
@@ -246,6 +246,28 @@ PyUFunc_ValidateCasting(PyUFuncObject *ufunc,
 }
 
 
+/*
+ * Same as `PyUFunc_ValidateCasting` but only checks output casting.
+ */
+NPY_NO_EXPORT int
+PyUFunc_ValidateOutCasting(PyUFuncObject *ufunc,
+        NPY_CASTING casting, PyArrayObject **operands, PyArray_Descr **dtypes)
+{
+    int i, nin = ufunc->nin, nop = nin + ufunc->nout;
+
+    for (i = nin; i < nop; ++i) {
+        if (operands[i] == NULL) {
+            continue;
+        }
+        if (!PyArray_CanCastTypeTo(dtypes[i],
+                PyArray_DESCR(operands[i]), casting)) {
+            return raise_output_casting_error(
```

Comment:
Yeah, it is a bit tricky, because this is only really used for the logical ufuncs, but the logical ufuncs throw this error already earlier in the legacy type resolution normally.  Maybe I will add a test for this using the scaled-float DType...

Reply:
Added a test, lets see what codecov thinks.